library/init/algebra/field.lean

```diff
@@ -16,9 +16,10 @@ set_option default_priority 100
 
 set_option old_structure_cmd true
 
-class division_ring (α : Type u) extends ring α, has_inv α, zero_ne_one_class α :=
+class division_ring (α : Type u) extends
+  add_group α, semigroup α, has_one α, distrib α, has_inv α, zero_ne_one_class α :=
```

Comment:
Why isn't a division ring a ring anymore? I don't see the proof below this

Reply:
There is a proof. `division_ring` doesn't require `one_mul`, but `ring` does. The proof of `one_mul` in a division ring is not entirely trivial, so the proof is some way down.